### PR TITLE
POC: Test for Helper Classes

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: tests/Base.php [hunk 1]
```diff
@@ -58,6 +58,24 @@ abstract class Base extends TestCase
...
+    ];
+
+    const PERMISSIONS_HELPER_RESPONSES = [
+        'read("any")',
+        'write("user:userid")',
+        'create("users")',
+        'update("guests")',
+        'delete("team:teamId/owner")',
+        'delete("team:teamId")',
+        'unique()',
```
Comment (abnegate) [inline tests/Base.php:76]: We should split this into `ID_HELPER_RESPONSES`

**Review by abnegate — CHANGES_REQUESTED** [review]

File: tests/Base.php (hunk 1 above)
Comment (christyjacob4) [reply tests/Base.php:79]: done 👍

File: tests/Android11Java11Test.php [hunk 2]
```diff
@@ -24,5 +24,8 @@ class Android11Java11Test extends Base
         ...Base::EXCEPTION_RESPONSES,
         ...Base::REALTIME_RESPONSES,
         ...Base::COOKIE_RESPONSES,
+        ...Base::QUERY_HELPER_RESPONSES,
+        ...BASE::PERMISSION_HELPER_RESPONSES,
+        ...BASE::ID_HELPER_RESPONSES
```
Comment (lohanidamodar) [inline tests/Android11Java11Test.php:29]:
```suggestion
        ...Base::PERMISSION_HELPER_RESPONSES,
        ...Base::ID_HELPER_RESPONSES
```

File: tests/Android11Java8Test.php [hunk 3]
```diff
@@ -24,5 +24,8 @@ class Android11Java8Test extends Base
         ...Base::EXCEPTION_RESPONSES,
         ...Base::REALTIME_RESPONSES,
         ...Base::COOKIE_RESPONSES,
+        ...Base::QUERY_HELPER_RESPONSES,
+        ...BASE::PERMISSION_HELPER_RESPONSES,
+        ...BASE::ID_HELPER_RESPONSES
```
Comment (lohanidamodar) [inline tests/Android11Java8Test.php:29]:
```suggestion
        ...Base::PERMISSION_HELPER_RESPONSES,
        ...Base::ID_HELPER_RESPONSES
```

File: tests/Android12Java11Test.php [hunk 4]
```diff
@@ -24,5 +24,8 @@ class Android12Java11Test extends Base
         ...Base::EXCEPTION_RESPONSES,
         ...Base::REALTIME_RESPONSES,
         ...Base::COOKIE_RESPONSES,
+        ...Base::QUERY_HELPER_RESPONSES,
+        ...BASE::PERMISSION_HELPER_RESPONSES,
+        ...BASE::ID_HELPER_RESPONSES
```
Comment (lohanidamodar) [inline tests/Android12Java11Test.php:29]:
```suggestion
        ...Base::PERMISSION_HELPER_RESPONSES,
        ...Base::ID_HELPER_RESPONSES
```

File: tests/Android12Java8Test.php [hunk 5]
```diff
@@ -24,5 +24,8 @@ class Android12Java8Test extends Base
         ...Base::EXCEPTION_RESPONSES,
         ...Base::REALTIME_RESPONSES,
         ...Base::COOKIE_RESPONSES,
+        ...Base::QUERY_HELPER_RESPONSES,
+        ...BASE::PERMISSION_HELPER_RESPONSES,
+        ...BASE::ID_HELPER_RESPONSES
```
Comment (lohanidamodar) [inline tests/Android12Java8Test.php:29]:
```suggestion
        ...Base::PERMISSION_HELPER_RESPONSES,
        ...Base::ID_HELPER_RESPONSES
```

File: tests/Android5Java11Test.php [hunk 6]
```diff
@@ -24,5 +24,8 @@ class Android5Java11Test extends Base
         ...Base::EXCEPTION_RESPONSES,
         ...Base::REALTIME_RESPONSES,
         ...Base::COOKIE_RESPONSES,
+        ...Base::QUERY_HELPER_RESPONSES,
+        ...BASE::PERMISSION_HELPER_RESPONSES,
+        ...BASE::ID_HELPER_RESPONSES
```
Comment (lohanidamodar) [inline tests/Android5Java11Test.php:29]:
```suggestion
        ...Base::PERMISSION_HELPER_RESPONSES,
        ...Base::ID_HELPER_RESPONSES
```

File: tests/Android5Java8Test.php [hunk 7]
```diff
@@ -24,5 +24,8 @@ class Android5Java8Test extends Base
         ...Base::EXCEPTION_RESPONSES,
         ...Base::REALTIME_RESPONSES,
         ...Base::COOKIE_RESPONSES,
+        ...Base::QUERY_HELPER_RESPONSES,
+        ...BASE::PERMISSION_HELPER_RESPONSES,
+        ...BASE::ID_HELPER_RESPONSES
```
Comment (lohanidamodar) [inline tests/Android5Java8Test.php:29]:
```suggestion
        ...Base::PERMISSION_HELPER_RESPONSES,
        ...Base::ID_HELPER_RESPONSES
```

File: tests/DartBetaTest.php [hunk 8]
```diff
@@ -19,5 +19,8 @@ class DartBetaTest extends Base
         ...Base::GENERAL_RESPONSES,
         ...Base::LARGE_FILE_RESPONSES,
         ...Base::EXCEPTION_RESPONSES,
+        ...Base::QUERY_HELPER_RESPONSES,
+        ...BASE::PERMISSION_HELPER_RESPONSES,
+        ...BASE::ID_HELPER_RESPONSES
```
Comment (lohanidamodar) [inline tests/DartBetaTest.php:24]:
```suggestion
        ...Base::PERMISSION_HELPER_RESPONSES,
        ...Base::ID_HELPER_RESPONSES
```

File: tests/DartStableTest.php [hunk 9]
```diff
@@ -19,5 +19,8 @@ class DartStableTest extends Base
         ...Base::GENERAL_RESPONSES,
         ...Base::LARGE_FILE_RESPONSES,
         ...Base::EXCEPTION_RESPONSES,
+        ...Base::QUERY_HELPER_RESPONSES,
+        ...BASE::PERMISSION_HELPER_RESPONSES,
+        ...BASE::ID_HELPER_RESPONSES
```
Comment (lohanidamodar) [inline tests/DartStableTest.php:24]:
```suggestion
        ...Base::PERMISSION_HELPER_RESPONSES,
        ...Base::ID_HELPER_RESPONSES
```

Seems this is problematic in all the tests, suggesting here is not efficient, please update the rest as well

**Review by lohanidamodar — COMMENTED** [review]
Some error in Response array in all the tests rest looks good to me.

**Pull request merged**